cli/src/pcluster/commands.py

```diff
@@ -445,3 +446,99 @@ def update(args):
+    try:
+        # delete image raises an exception if stack does not exist
+        result = PclusterApi().delete_image(image_name=args.image_name, region=utils.get_region(), force=args.force)
+        if isinstance(result, ImageBuilderInfo):
+            print(f"Image deletion started correctly. {result}")
+        else:
+            utils.error(f"Image deletion failed. {result.message}")
+
+        if result.stack_exist:
+            sys.stdout.write("\rImageBuilderStackStatus: %s" % result.stack_status)
+            sys.stdout.flush()
+            LOGGER.debug("ImageBuilderStackStatus: %s", result.stack_status)
+
+            if not args.nowait:
+                while result.stack_status == "DELETE_IN_PROGRESS":
+                    time.sleep(1)
+                    result = PclusterApi().describe_image(image_name=args.image_name, region=utils.get_region())
+                    if isinstance(result, ImageBuilderInfo) and result.stack_exist:
+                        events = utils.get_stack_events(result.stack_name, raise_on_error=True)[0]
+                        resource_status = (
+                            "Status: %s - %s" % (events.get("LogicalResourceId"), events.get("ResourceStatus"))
+                        ).ljust(80)
+                        sys.stdout.write("\r%s" % resource_status)
+                        sys.stdout.flush()
+                    elif isinstance(result, ApiFailure) or not result.stack_exist:
+                        # If stack is already deleted
+                        if f"ImageBuilder stack {args.image_name} does not exist." in result.message or (
+                            isinstance(result, ImageBuilderInfo) and not result.stack_exist
+                        ):
+                            LOGGER.warning(f"\nImage {args.image_name} has already been deleted or does not exist.")
+                            sys.exit(0)
+                        LOGGER.critical(result.message)
+                        sys.stdout.flush()
+                        sys.exit(1)
+                    else:
+                        utils.error(f"Unable to retrieve the status of the image.\n{result.message}")
+                sys.stdout.write("\rStatus: %s\n" % result.stack_status)
+                sys.stdout.flush()
+                LOGGER.debug("Status: %s", result.stack_status)
+            else:
+                sys.stdout.write("\n")
+                sys.stdout.flush()
+            if result.stack_status == "DELETE_FAILED":
+                LOGGER.info("Image did not delete successfully. Run 'pcluster delete-image %s' again", args.image_name)
+
+        else:
+            sys.stdout.write("\rImageStatus: %s" % result.image_state)
+            sys.stdout.flush()
+            LOGGER.debug("ImageStatus: %s", result.image_state)
+
+            if not args.nowait:
+                image_name = result.image_name
+                while result.image_exist:
+                    time.sleep(0.1)
```

Comment:
Is 0.1 the right value?

Reply:
deregistering image and deleting snapshot is super quick, I think nowait is meanless if it is the case of image deletion not stack deletion. I'd remove nowait logic if it is image deletion.